CRM/Upgrade/Incremental/General.php

```diff
@@ -77,6 +84,27 @@ public static function setPreUpgradeMessage(&$preUpgradeMessage, $currentVer, $l
       ]);
       $preUpgradeMessage .= '</p>';
     }
+    if (version_compare(CRM_Utils_SQL::getDatabaseVersion(), self::MIN_RECOMMENDED_MYSQL_VER) < 0) {
+      $preUpgradeMessage .= '<p>';
+      $preUpgradeMessage .= ts('You may proceed with the upgrade and CiviCRM %1 will continue working normally, but future releases will require MySQL version %2 or above or MariaDB version %3 or above. We recommend PHP version %4.', [
+        1 => $latestVer,
+        2 => self::MIN_RECOMMENDED_MYSQL_VER,
+        3 => '10.1',
+        4 => preg_replace(';^(\d+\.\d+(?:\.[1-9]\d*)?).*$;', '\1', self::RECOMMENDED_MYSQL_VER),
+      ]);
+      $preUpgradeMessage .= '</p>';
+    }
+    if (version_compare(CRM_Utils_SQL::getDatabaseVersion(), self::NEW_MIN_INSTALL_MYSQL_VER) < 0) {
+      $preUpgradeMessage .= '<p>';
+      $preUpgradeMessage .= ts('You may proceed with the upgrade and CiviCRM %1 will continue working normally, but CiviCRM versions from 5.28 onwards will require MySQL version %1 at least it is recommended to use MySQL version %3 or above or MariaDB version %4 or above. We recommend PHP version %5.', [
```

Comment:
There is just a wording issue around "require MySQL version %1 at least it is "

Maybe 
at least MySql version %1. It is....